FlowCal/plot.py

```diff
@@ -1058,21 +1141,33 @@ def scatter2d(data_list,
         plt.ylabel(data_plot.channels[1])
 
     # Set scale of axes
-    plt.gca().set_xscale(xscale)
-    plt.gca().set_yscale(yscale)
+    plt.gca().set_xscale(xscale, data=data_list, channel=channels[0])
+    plt.gca().set_yscale(yscale, data=data_list, channel=channels[1])
 
-    # Set plot limits if specified, else extract range from data_plot
+    # Set plot limits if specified, else extract range from data_list
     # Use data_plot.hist_bins with one single bin
-    if xlim is not None:
-        plt.xlim(xlim)
-    elif hasattr(data_plot, 'hist_bins') and \
-            hasattr(data_plot.hist_bins, '__call__'):
-        plt.xlim(data_plot.hist_bins(channels=0, nbins=1, scale=xscale))
-    if ylim is not None:
-        plt.ylim(ylim)
-    elif hasattr(data_plot, 'hist_bins') and \
-            hasattr(data_plot.hist_bins, '__call__'):
-        plt.ylim(data_plot.hist_bins(channels=1, nbins=1, scale=yscale))
+    if xlim is None:
+        xlim = [np.inf, -np.inf]
+        for data in data_list:
+            if hasattr(data, 'hist_bins') and \
+                    hasattr(data.hist_bins, '__call__'):
+                xlim_data = data.hist_bins(channels=channels[0],
+                                           nbins=1,
+                                           scale=xscale)
```

Comment:
Another issue is that floating-point files have negative events, which will not be covered by `.range` independently of whether we start with 0 or 1. Logicle axis adapt to this, but if we manually force the lower axis limit to be equal to the lower range value, we will not show negative values and the whole point of logicle scaling is lost.

In short, the range is not very precise, and `hist_bins` does a better job at giving limits for visualization.